vllm/engine/async_llm_engine.py

```diff
@@ -581,6 +581,7 @@ async def add_request_async(
             lora_request=lora_request,
             prompt_adapter_request=prompt_adapter_request,
         )
+        self._validate_model_inputs(processed_inputs)
```

Comment:
@maxdebayser move inside the `_add_processed_request` method below? Then the check is only in one place since this is used by both LLMEngine and AsyncLLMEngine.